### PR TITLE
feat: estimate query cost in Postgres

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Currently the estimate query cost feature is only enabled for Presto, making it hard to test the functionality and iterate on SQL Lab without breaking it.

This PR adds a simple query cost estimator to Postgres.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screenshot_2020-12-18 Superset

### TEST PLAN

See screenshot.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/presto.py [hunk 1]
```diff
@@ -132,7 +132,8 @@ class PrestoEngineSpec(BaseEngineSpec):  # pylint: disable=too-many-public-metho
     }
 
     @classmethod
-    def get_allow_cost_estimate(cls, version: Optional[str] = None) -> bool:
+    def get_allow_cost_estimate(cls, extra: Dict[str, Any]) -> bool:
+        version = extra.get("version")
         return version is not None and StrictVersion(version) >= StrictVersion("0.319")
```
Comment (hughhhh) [inline superset/db_engine_specs/presto.py:137]: one liner:
`return extra.get("version") and StrictVersion(version) >= StrictVersion("0.319")`

**Review by hughhhh — APPROVED** [review]

**Comment by ktmud** [comment]
What's the unit of this cost? Seconds?

**Comment by betodealmeida** [comment]
> What's the unit of this cost? Seconds?

The costs are measured in arbitrary units determined by the planner's cost parameters (see Section 18.7.2). Traditional practice is to measure the costs in units of disk page fetches; that is, seq_page_cost is conventionally set to 1.0 and the other cost parameters are set relative to that. The examples in this section are run with the default cost parameters. [[reference](https://www.postgresql.org/docs/9.4/using-explain.html)]

It's a relative value, so not super useful. But it's possible to collect stats about queries, and define a custom `QUERY_COST_FORMATTERS_BY_ENGINE` that maps from the relative values to something else. For example, you could run `EXPLAIN` every day in all recent queries, and use the results to built a histogram, so that the formatter says something like "this query is at the top 3% percentile of all queries".

**Comment by ktmud** [comment]
Thanks for the explanation! In that case, do you think it's worth adding the link to Postgres doc somewhere in the modal. Or at least a one-liner to explain how is this calculated? E.g.

```
Query cost is estimated by `EXPLAIN` statement. Refer to the database docs for details.
```

**Comment by betodealmeida** [comment]
@ktmud unfortunately the modal is not DB-specific, so we can't add that message to it.

I also don't expect people to use it without an admin setting up a custom formatter that translates the number into a meaningful value. I added an example to `config.py` showing to set it up so that the numbers are more tangible.

Here's a screenshot with that config in place:

Screen Shot 2020-12-18 at 3 13 56 PM

**Comment by ktmud** [comment]
@betodealmeida Thanks for the explanation. This makes sense.

**Pull request merged**